src/Bicep.Core.Samples/DataSetsExtensions.cs

```diff
@@ -147,32 +161,38 @@ public static ITemplateSpecRepositoryFactory CreateMockTemplateSpecRepositoryFac
         }
 
         public static async Task PublishModulesToRegistryAsync(this DataSet dataSet, IContainerRegistryClientFactory clientFactory)
+        {
+            foreach (var (moduleName, publishInfo) in dataSet.RegistryModules)
+            {
+                await PublishModuleToRegistryAsync(clientFactory, moduleName, publishInfo.Metadata.Target, publishInfo.ModuleSource, null);
```

Comment:
Should `Task::WhenAll` or `WhenAllForAwait` be used here?

Reply:
Perhaps, but this is existing test code (just refactored), and I think not worth the risk or the low benefit (mocks).